src/persistence/serialize.cpp

```diff
@@ -227,52 +95,29 @@ size_t dataToVUint(const QByteArray& data)
 unsigned getVUint32Size(QByteArray data)
 {
     unsigned lensize=0;
-    {
-        unsigned char num3;
-        do {
-            num3 = data[lensize];
-            lensize++;
-        } while ((num3 & 0x80) != 0);
-    }
-    return lensize;
-}
 
-QByteArray uint8ToData(uint8_t num)
-{
-    QByteArray data(1,0);
-    data[0] = (uint8_t)num;
-    return data;
-}
-
-QByteArray uint16ToData(uint16_t num)
-{
-    QByteArray data(2,0);
-    data[0] = (uint8_t)(num & 0xFF);
-    data[1] = (uint8_t)((num>>8) & 0xFF);
-    return data;
-}
+    char num3;
+    do {
+        num3 = data[lensize];
+        lensize++;
+    } while ((num3 & 0x80) != 0);
 
-QByteArray uint32ToData(uint32_t num)
-{
-    QByteArray data(4,0);
-    data[0] = (uint8_t)(num & 0xFF);
-    data[1] = (uint8_t)((num>>8) & 0xFF);
-    data[2] = (uint8_t)((num>>16) & 0xFF);
-    data[3] = (uint8_t)((num>>24) & 0xFF);
-    return data;
+    return lensize;
 }
 
-QByteArray uint64ToData(uint64_t num)
+QByteArray vintToData(int num)
 {
-    QByteArray data(8,0);
-    data[0] = (uint8_t)(num & 0xFF);
-    data[1] = (uint8_t)((num>>8) & 0xFF);
-    data[2] = (uint8_t)((num>>16) & 0xFF);
-    data[3] = (uint8_t)((num>>24) & 0xFF);
-    data[4] = (uint8_t)((num>>32) & 0xFF);
-    data[5] = (uint8_t)((num>>40) & 0xFF);
-    data[6] = (uint8_t)((num>>48) & 0xFF);
-    data[7] = (uint8_t)((num>>56) & 0xFF);
+    QByteArray data(sizeof(int), 0);
+    // Write the size in a Uint of variable lenght (8-32 bits)
+    int i=0;
+    while (num >= 0x80)
+    {
+        data[i] = static_cast<char>(num | 0x80);
+        i++;
```

Comment:
Yep, for some cases and some compilers it may matter, but we are writing for many compilers. Despite that, premature optimization kills the program.